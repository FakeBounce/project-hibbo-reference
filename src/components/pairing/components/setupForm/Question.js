import React from 'react';
import { Animated } from 'react-native';
import { bool, number, string, func, shape, instanceOf } from 'prop-types';

import { StyledContainerBasic } from 'styledComponents/containers';

import appStyles from 'styles/appStyles';

import Title from '../Title';

import * as Styles from './styles';

const Question = ({
  step,
  displayTitle,
  title,
  controls,
  setData,
  moveStep,
  redirect,
  titleAnimations,
  controlsAnimations,
  showContent,
}) => {
  return (
    <StyledContainerBasic>
      {displayTitle && (
        <Title
          label={[title.label1, title.label2, title.label3, title.label4]}
          color={appStyles.colors.white}
          animations={titleAnimations}
          showContent={showContent}
        />
      )}

      <Styles.SControls
        leftLabel={controls.leftLabel}
        leftAction={() => {
          if (step === 3) {
            return setData('gender', 'female', 'left');
          } else if (step === 8) {
            return redirect(false, false, 'left');
          }

          return moveStep();
        }}
        rightLabel={controls.rightLabel}
        rightAction={() => {
          if (step === 3) {
            return setData('gender', 'male', 'right');
          } else if (step === 8) {
            return redirect(true, true, 'right');
          }

          return moveStep();
        }}
        animations={controlsAnimations}
      />
    </StyledContainerBasic>
  );
};

Question.defaultProps = {
  displayTitle: true,
  title: {
    label1: '',
    label2: '',
    label3: '',
  },
  controls: {
    rightLabel: '',
    leftLabel: '',
  },
  showContent: null,
};

Question.propTypes = {
  step: number.isRequired,
  displayTitle: bool,
  title: shape({
    label1: string,
    label2: string,
    label3: string,
  }),
  controls: shape({
    leftLabel: string,
    rightLabel: string,
  }),
  setData: func.isRequired,
  moveStep: func.isRequired,
  redirect: func.isRequired,
  titleAnimations: shape({
    opacity: instanceOf(Animated.Value).isRequired,
    translateY: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  controlsAnimations: shape({
    left: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
    right: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
  }).isRequired,
  showContent: func,
};

export default Question;
