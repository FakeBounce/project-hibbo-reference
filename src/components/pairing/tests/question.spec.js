import React from 'react';
import { Animated } from 'react-native';
import renderer from 'react-test-renderer';

import Question from '../components/setupForm/Question';

describe('Question', () => {
  const fn = jest.fn();
  const title = {
    label1: '',
    label2: '',
    label3: '',
  };
  const controls = {
    rightLabel: '',
    leftLabel: '',
  };
  const animatedStyles = {
    titleOpacity: new Animated.Value(1),
    buttonOpacity: new Animated.Value(1),
  };
  const form = {
    name: '',
    picture: '',
  };
  const navigation = {
    navigate: fn,
  };
  const component = renderer.create(
    <Question
      step={1}
      setData={fn}
      validate={fn}
      error={false}
      title={title}
      controls={controls}
      animatedStyles={animatedStyles}
      form={form}
      navigation={navigation}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
