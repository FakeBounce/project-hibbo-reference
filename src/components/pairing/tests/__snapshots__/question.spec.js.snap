// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render the component 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          initialDelay={200}
          maxDelay={100}
          minDelay={20}
          onTypingEnd={[Function]}
          style={
            Array [
              undefined,
              Object {
                "color": "#FFDF2D",
              },
            ]
          }
          typing={1}
        >
          
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          initialDelay={200}
          maxDelay={100}
          minDelay={20}
          onTypingEnd={[Function]}
          style={
            Array [
              undefined,
              Object {
                "color": "#FFDF2D",
              },
            ]
          }
          typing={0}
        >
          
        </Text>
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "bottom": 25,
        "opacity": 1,
        "position": "absolute",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
  </View>
</View>
`;
